### PR TITLE
adding in files to the work class

### PR DESCRIPTION
**Pull request by cam156**
## Description

Add in File upload for a work.  This just attaches singular files to a work.  There no components for displaying the contents of the file, downloading the file, or deleting the file.

References ticket: (if any) #359 

Why was this necessary?

## Changes

* added in shrine gem
* added Work::File object
* added file to the Submission::ChangeSet
* added file persistence to the ChangeSetPersister
* added html5 input to the edit and new screen
* added display of attached files to the work show screen

Are there any major changes in this PR that will change the release process? No

## Checklist

- [x] i18n enabled
- [x] adequate test coverage
- [x] accessible interface

### CODE REVIEW

File: app/cho/work/submission.rb [hunk 1]
```diff
@@ -8,5 +8,8 @@ class Submission < Valkyrie::Resource
 
     attribute :id, Valkyrie::Types::ID.optional
     attribute :work_type, Valkyrie::Types::String
+    attribute :files, Valkyrie::Types::Set
+
+    attr_accessor :file_data, :cached_file_data, :file
```
Comment (awead) [inline app/cho/work/submission.rb]: Could you comment more on what these are? I'm assuming they're from Shrine. I'm also confused about `files` versus `file`. The attributes is multiple, but there's a `file` accessor... how does that differ?

File: app/cho/work/submission_change_set.rb [hunk 2]
```diff
@@ -4,6 +4,7 @@ module Work
   class SubmissionChangeSet < Valkyrie::ChangeSet
     validates :work_type, presence: true
     property :work_type, multiple: false, required: true
+    property :file, multiple: false, required: false
```
Comment (awead) [inline app/cho/work/submission_change_set.rb]: Why is the change set singular while the model is multiple? Are we only uploading one file at a time?

File: app/cho/work/file.rb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+# frozen_string_literal: true
+
+class Work::File < Valkyrie::Resource
+  include CommonQueries
+
+  attribute :id, Valkyrie::Types::ID.optional
+  attribute :original_filename, Valkyrie::Types::String
+  attribute :use, Valkyrie::Types::Set
+  attribute :file_identifier, Valkyrie::Types::ID.optional
```
Comment (awead) [inline app/cho/work/file.rb]: Is this another UUID-type identifier for the file, or something different?

File: app/cho/work/submission_change_set.rb (hunk 2 above)
Comment (cam156) [reply app/cho/work/submission_change_set.rb]: @awead yes at this point I only set up shrine to allow for one file

File: app/cho/work/file.rb (hunk 3 above)
Comment (cam156) [reply app/cho/work/file.rb]: @awead that is the actual binary content coming out of storage

Comment (awead) [reply app/cho/work/file.rb]: @cam156 so if I call `resource.file_identifier` I get binary data back? That seems odd given that the field type is an id. If this is a Shrine/Valkyrie thing, that's fine, but what about a better attribute name such as `content` or `file_handle`?

Comment (cam156) [reply app/cho/work/file.rb]: @awead it is an identifier for the storage object, so you still need to call find.  I took this concept from valkyrie: https://github.com/samvera-labs/valkyrie/blob/master/spec/valkyrie/derivative_service_spec.rb

**Pull request merged**